src/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -244,14 +245,15 @@ internal void InternalPreserveStackTrace()
             _stackTraceString = null;
         }
 
-
-        // This is the object against which a lock will be taken
-        // when attempt to restore the EDI. Since its static, its possible
-        // that unrelated exception object restorations could get blocked
-        // for a small duration but that sounds reasonable considering
-        // such scenarios are going to be extremely rare, where timing
-        // matches precisely.
-        private static readonly object s_DispatchStateLock = new object();
+        // This is the object against which a lock will be taken when attempting to restore the EDI
+        // rather than using it directly, the property DispatchStateLock should be used.
+        private object? _dispatchStateLock;
+        private object DispatchStateLock => _dispatchStateLock ?? CreateDispatchLock();
+        private object CreateDispatchLock()
```

Comment:
This is not hot enough to deserve this outlining micro-optimization.

Reply:
Removed